index.js

```diff
@@ -205,16 +280,14 @@ jasmine.Expectation.prototype.wrapCompare = function(name, matcherFactory) {
 
     matchError.stack = matchError.stack.replace(/ +at.+jasminewd.+\n/, '');
 
-    if (!webdriver.promise.isPromise(expectation.actual) &&
-        !webdriver.promise.isPromise(expected)) {
-      compare(expectation.actual, expected);
-    } else {
-      webdriver.promise.when(expectation.actual).then(function(actual) {
-        return webdriver.promise.all(expected).then(function(expected) {
-          return compare(actual, expected);
-        });
+    expectation.actual = extractPromiseFromDeferred(expectation.actual);
```

Comment:
For the record: we looked into this offline, and we can't figure out why that change was originally made. As far as we can tell, there's no reason anyone should do `expect(foo)...` where `foo` is a deferred, not a promise.

In `selenium-webdriver@2`, all deferred objects _were_ promises. That's changed with version 3, and is a breaking change, but we don't expect this to affect very many users at all (and it will be released in a major version).

So we're going to remove the special logic to handle deferreds,